pkg/applicationserver/applicationserver.go

```diff
@@ -588,6 +592,202 @@ func (as *ApplicationServer) handleJoinAccept(ctx context.Context, ids ttnpb.End
 	return nil
 }
 
+// downlinkQueuesUpdater creates the pending and current session downlink queues.
+// The updater generally mutates the end device's sessions.
+type downlinkQueuesUpdater func(context.Context, *ttnpb.EndDevice) ([]*ttnpb.ApplicationDownlink, []*ttnpb.ApplicationDownlink, error)
+
+// changeDownlinkQueue updates the downlink queue of the end device with the given downlink queues updater.
```

Comment:
Change vs update

It's not clear to me what really happens here from this naming and description